### PR TITLE
[IMP][14.0] Module: base_user_role show rules and access

### PR DESCRIPTION
**Pull request by Kev-Roche**
Add smart buttons for access rights and record rules linked to a role, on role view.

### CODE REVIEW

**Comment by Kev-Roche** [comment]
cc @sebalix, @jcdrubay, @novawish, @sebastienbeau 
What do you think about this improvement?

File: base_user_role/models/role.py [hunk 1]
```diff
@@ -39,6 +57,20 @@ def _compute_user_ids(self):
         for role in self:
             role.user_ids = role.line_ids.mapped("user_id")
 
+    @api.depends("implied_ids", "implied_ids.model_access")
+    def _compute_model_access_ids(self):
+        for rec in self:
+            access = rec.implied_ids.model_access.ids
+            rec.model_access_ids = [(6, 0, access)]
+            rec.model_access_count = len(access)
```
Comment (sebastienbeau) [inline base_user_role/models/role.py]: shorter (yoiu can do the same for rules)
```suggestion
            rec.model_access_ids = rec.implied_ids.model_access.ids
            rec.model_access_count = len(rec.model_access_ids)
```

**Review by sebastienbeau — CHANGES_REQUESTED** [review]
minor change, good improvement

File: base_user_role/models/role.py (hunk 1 above)
Comment (Kev-Roche) [reply base_user_role/models/role.py]: done

File: base_user_role/models/role.py [hunk 2]
```diff
@@ -26,6 +26,24 @@ class ResUsersRole(models.Model):
...
+
+    model_access_ids = fields.Many2many(
+        comodel_name="ir.model.access",
+        compute="_compute_model_access_ids",
+        string="Access Rights",
+        required=False,
+    )
+
+    model_access_count = fields.Integer(compute="_compute_model_access_ids")
+
```
Comment (sebalix) [inline base_user_role/models/role.py]: No need to put empty lines between field declarations

File: base_user_role/models/role.py [hunk 3]
```diff
@@ -75,6 +105,30 @@ def cron_update_users(self):
...
+        return {
+            "name": "Record Rules",
+            "type": "ir.actions.act_window",
+            "res_id": self.id,
+            "view_mode": "tree",
+            "res_model": "ir.rule",
+            "view_id": self.env.ref("base.view_rule_tree").id,
+            "target": "current",
+            "domain": [("id", "in", self.rule_ids.ids)],
+        }
```
Comment (sebalix) [inline base_user_role/models/role.py]: Could we reuse the existing action and tune it instead (we avoid to translate new terms this way)?
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("base.action_rule")
        action["domain"] = [("id", "in", self.rule_ids.ids)]
        return action
```

File: base_user_role/models/role.py [hunk 4]
```diff
@@ -75,6 +105,30 @@ def cron_update_users(self):
...
+            "name": "Access Rights",
+            "type": "ir.actions.act_window",
+            "res_id": self.id,
+            "view_mode": "tree",
+            "res_model": "ir.model.access",
+            "view_id": self.env.ref("base.ir_access_view_tree_edition").id,
+            "target": "current",
+            "domain": [("id", "in", self.model_access_ids.ids)],
+        }
+
```
Comment (sebalix) [inline base_user_role/models/role.py]: Same remark:
```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("base.ir_access_act")
        action["domain"] = [("id", "in", self.model_access_ids.ids)]
        return action
```

**Review by sebalix — CHANGES_REQUESTED** [review]
Nice addition, thank you.
But you have to remove all changes related to PO files and do them through Weblate (https://translation.odoo-community.org/ ) to keep it synchronized as this module already exists in the 14.0 branch.

**Comment by Kev-Roche** [comment]
> Nice addition, thank you. But you have to remove all changes related to PO files and do them through Weblate (https://translation.odoo-community.org/ ) to keep it synchronized as this module already exists in the 14.0 branch.

@sebalix 
Thank you for your time review, I add your improvements.

**Review by sebalix — APPROVED** [review]

**Review by TDu — COMMENTED** [review]
LG, only comment would be not to have version number in the commit message.

**Review by TDu — APPROVED** [review]

**Comment by sebalix** [comment]
Good catch, @Kev-Roche can you rewrite the commit message without the version number as this one will be ported to other branches? Like `[IMP] base_user_role: show rules and access`
I'll merge it right after.

**Comment by Kev-Roche** [comment]
@TDu , @sebalix commit message changed.

**Comment by sebalix** [comment]
@Kev-Roche thank you for the updates!
/ocabot merge minor

**Pull request merged**